### PR TITLE
feat(kraken): add pagination to fetchWithdrawals

### PR DESCRIPTION
**Pull request by pcriadoperez**
- Add parsing for withdrawMethods networks
- Remove required code
- Add cursor pagination to fetchWithdrawals

### CODE REVIEW

**Comment by pcriadoperez** [comment]
### No pagination and no code
```
kraken.fetchWithdrawals(None,None,None)
[{'address': 'TBt85AZ29oXatt1MHP*******************',
  'addressFrom': None,
  'addressTo': None,
  'amount': 4.0,
  'comment': None,
  'currency': 'USDT',
  'datetime': '2023-12-01T21:09:23.000Z',
  'fee': {'cost': 3.0, 'currency': 'USDT'},
  'id': 'FTQWjwJ-tFpFqqSfp50HHmVTiqIR1a',
  'info': {'aclass': 'currency',
           'amount': '4.00000000',
           'asset': 'USDT',
           'fee': '3.00000000',
           'info': 'TBt85AZ29oXatt1MHPPna*************************',
           'key': 'okx',
           'method': 'Tether USD (TRC20)',
           'network': 'Tron',
           'refid': 'FTQWjwJ-tFpFqqSfp50HHmVTiqIR1a',
           'status': 'Success',
           'time': '1701464963',
           'txid': '7a5248142ebfceddf7c4fec2ea59ff845ece4de1d1e3e7a504d1f46666a6d706',
           'type': 'withdrawal'},
  'internal': None,
  'network': 'TRC20',
  'status': 'ok',
  'tag': None,
  'tagFrom': None,
  'tagTo': None,
  'timestamp': 1701464963000,
  'txid': '7a5248142ebfceddf7c4fec2ea59ff845ece4de1d1e3e7a504d1f46666a6d706',
  'type': 'withdrawal',
  'updated': None},
 {'address': 'TFqAnCgT6aHJNUUWG*******************',
  'addressFrom': None,
  'addressTo': None,
  'amount': 3.0,
  'comment': None,
  'currency': 'USDT',
  'datetime': '2023-12-01T21:08:55.000Z',
  'fee': {'cost': 3.0, 'currency': 'USDT'},
  'id': 'FTWtk9D-3ksJzJRN5NgIGdl9JC6BOh',
  'info': {'aclass': 'currency',
           'amount': '3.00000000',
           'asset': 'USDT',
           'fee': '3.00000000',
           'info': 'TFqAnCgT6aHJNUUWGM*******************',
           'key': 'Huobi wallet',
           'method': 'Tether USD (TRC20)',
           'network': 'Tron',
           'refid': 'FTWtk9D-3ksJzJRN5NgIGdl9JC6BOh',
           'status': 'Success',
           'time': '1701464935',
           'txid': 'a11b4f23d054d7061356d986688a4ed474cf3a3d195f108fd2fab897a5853271',
           'type': 'withdrawal'},
  'internal': None,
  'network': 'TRC20',
  'status': 'ok',
  'tag': None,
  'tagFrom': None,
  'tagTo': None,
  'timestamp': 1701464935000,
  'txid': 'a11b4f23d054d7061356d986688a4ed474cf3a3d195f108fd2fab897a5853271',
  'type': 'withdrawal',
  'updated': None}]
  ```
  
  ### with pagination and since
  ```python
  kraken.fetchWithdrawals(None,1666166569000,None,{'paginate': 'True'})
[{'address': 'n26',
  'addressFrom': None,
  'addressTo': None,
  'amount': 998.91,
  'comment': None,
  'currency': 'EUR',
  'datetime': '2022-10-19T08:02:49.000Z',
  'fee': {'cost': 0.09, 'currency': 'EUR'},
  'id': 'ACCLAI4-WK5KXP-B7CVNQ',
  'info': {'aclass': 'currency',
           'amount': '998.9100',
           'asset': 'ZEUR',
           'fee': '0.0900',
           'info': 'n26',
           'key': 'N26 Conta',
           'method': 'SEPA',
           'refid': 'ACCLAI4-WK5KXP-B7CVNQ',
           'status': 'Success',
           'time': '1666166569',
           'txid': '248620356',
           'type': 'withdrawal'},
  'internal': None,
  'network': None,
  'status': 'ok',
  'tag': None,
  'tagFrom': None,
  'tagTo': None,
  'timestamp': 1666166569000,
  'txid': '248620356',
  'type': 'withdrawal',
  'updated': None},
  ```

**Comment by pcriadoperez** [comment]
```php
kraken->fetchWithdrawals()
Array
(
    [0] => Array
        (
            [info] => Array
                (
                    [type] => withdrawal
                    [method] => Tether USD (TRC20)
                    [aclass] => currency
                    [asset] => USDT
                    [refid] => FTQWjwJ-tFpFqqSfp50HHmVTiqIR1a
                    [txid] => 7a5248142ebfceddf7c4fec2ea59ff845ece4de1d1e3e7a504d1f46666a6d706
                    [info] => TBt85AZ29oXa****************************
                    [amount] => 4.00000000
                    [fee] => 3.00000000
                    [time] => 1701464963
                    [status] => Success
                    [key] => okx
                    [network] => Tron
                )

            [id] => FTQWjwJ-tFpFqqSfp50HHmVTiqIR1a
            [currency] => USDT
            [amount] => 4
            [network] => TRC20
            [address] => TBt85AZ29oXatt1MHPPn************
            [addressTo] => 
            [addressFrom] => 
            [tag] => 
            [tagTo] => 
            [tagFrom] => 
            [status] => ok
            [type] => withdrawal
            [updated] => 
            [txid] => 7a5248142ebfceddf7c4fec2ea59ff845ece4de1d1e3e7a504d1f46666a6d706
            [timestamp] => 1701464963000
            [datetime] => 2023-12-01T21:09:23.000Z
            [comment] => 
            [internal] => 
            [fee] => Array
                (
                    [currency] => USDT
                    [cost] => 3
                )

        )

    [1] => Array
    ...
    ```

File: ts/src/kraken.ts [hunk 1]
```diff
@@ -325,6 +325,91 @@ export default class kraken extends Exchange {
                     'ZEC': 'Zcash (Transparent)',
                     'ZRX': '0x (ZRX)',
                 },
+                'withdrawMethods': {
```
Comment (carlosmiei) [inline ts/src/kraken.ts:328]: @pcriadoperez two questions, first why aren't these inside `networks` but the other way around (UNIFIED:ID)? By doing that `networksById` is computed automatically.


Second, why not use `networkIdToCode/networkCodeToID` to parse these values?

Comment (carlosmiei) [reply ts/src/kraken.ts:328]: > found some differences between the naming of the same network depending if they are deposits or withdrawals

@pcriadoperez damn that's bad 😅  I don't get why they would do things like that

**Pull request merged**